x-pack/plugins/lens/public/datasources/form_based/operations/definitions/formula/util.ts

```diff
@@ -787,3 +790,26 @@ export function filterByVisibleOperation(
     Object.entries(operationDefinitionMap).filter(([, operation]) => !operation.hidden)
   );
 }
+
+export function isCounterRateFormula(col: GenericIndexPatternColumn, indexPattern: IndexPattern) {
+  const formula = (col as FormulaIndexPatternColumn).params.formula || '';
+  const ast = parse(formula);
+  if (typeof ast !== 'number' && ast.type === 'function') {
+    const { name, args } = ast;
+    if (
+      name === 'counter_rate' &&
+      typeof args[0] !== 'number' &&
+      args[0].type === 'function' &&
+      args[0].name === 'max' &&
```

Comment:
shouldn't this be a list of supported metric?

Right now if I pick any other operation other than `max` for a counter field I get this unclear error message:
<img width="371" alt="Screenshot 2023-04-05 at 14 39 52" src="https://user-images.githubusercontent.com/924948/230083989-01a2a818-f6fe-4260-9cc9-d391f3ec4698.png">

Or are you waiting for field caps to return for each field the supported agg list?

Reply:
average is not supported on counter field as far as i know, you can use min and max for now, possibly we'll add more later. But only when using max we'll use es rate aggregation (as that's the calculation it would use internally), with min we'll keep doing the counter rate operation in lens.